### ISSUE TITLE
Advice docs are too verbose

### ISSUE THREAD

**Issue by allan2** [RazrFalcon/memmap2-rs#134]
The docs in the advice module are long. They are mostly copied and pasted from the [Linux man page](https://man7.org/linux/man-pages/man2/madvise.2.html).

Some current problems:
- There are unlinked references to man pages such as `fork(2)`.
- The formatting is copied over as well, such as two spaces after a sentence period.
- The sheer length makes it hard to read _advice.rs_.
- The information is not always up to date. #120

I propose limiting each value's comment to just a few lines. A brief explanation and a safety block when needed. The `Advice` enum's comment already includes a link to the man page, for those who need more explanation.

madvise is also not just for Linux. Other reference pages include [FreeBSD](https://man.freebsd.org/cgi/man.cgi?query=madvise&sektion=2) and [Mac OS X](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/madvise.2.html).

#133  affected

**Comment by de-vri-es** [comment]
Agreed. If we can simplify the text here and refer to external documentation for platform specific details, I think that's better.
